code/mapping/ext_modules/mapping_common/transform.py

```diff
@@ -0,0 +1,305 @@
+            - angle < 0: CW
+        """
+        # Based on https://stackoverflow.com/questions/21483999/
+        # using-atan2-to-find-angle-between-two-vectors/21486462#21486462
+        cross = np.cross(self._matrix[:2], other._matrix[:2])
+        dot = np.dot(self._matrix[:2], other._matrix[:2])
+        return math.atan2(cross, dot)
+
+    def point(self) -> Point2:
+        return Point2(self._matrix)
+
+    @staticmethod
+    def new(x: float, y: float) -> "Vector2":
+        m = np.array([x, y, 1.0], dtype=np.float64)
+        return Vector2(m)
+
+    @staticmethod
+    def zero() -> "Vector2":
+        return Vector2.new(0.0, 0.0)
+
+    @staticmethod
+    def from_point(p: Point2) -> "Vector2":
+        return Vector2(p._matrix)
+
+    @staticmethod
+    def from_ros_msg(m: geometry_msgs.Vector3) -> "Vector2":
+        return Vector2.new(m.x, m.y)
+
+    def to_ros_msg(self) -> geometry_msgs.Vector3:
+        return geometry_msgs.Vector3(x=self.x(), y=self.y(), z=0.0)
+
+    def __mul__(self, other):
+        if isinstance(other, float):
+            matrix = self._matrix * other
+            matrix[2] = 1.0
+            return Vector2(matrix)
+        raise TypeError(
+            f"Unsupported operand types for *: '{type(self)}' and '{type(other)}'"
+        )
+
+    def __add__(self, other):
+        if isinstance(other, Vector2):
+            matrix = self._matrix + other._matrix
+            matrix[2] = 1.0
+            return Vector2(matrix)
+        raise TypeError(
+            f"Unsupported operand types for *: '{type(self)}' and '{type(other)}'"
+        )
+
+
+@dataclass(init=False, eq=False)
+class Transform2D:
+    """Homogeneous 2 dimensional transformation matrix
+
+    Based on https://alexsm.com/homogeneous-transforms/
+
+    ## Examples:
+    ### Transform a Vector2
+    ```python
+    v = Vector2.new(1.0, 0.0)
+    t = Transform2D.new_rotation(math.pi/2.0)
+    v_transformed = t * v
+    ```
+    v_transformed is (0.0, 1.0)
+
+    Note that Vectors are only directions/offsets and ignore translations.
+    """
+
+    # Matrix with shape (3, 3)
+    _matrix: npt.NDArray[np.float64]
+
+    def __init__(self, matrix: npt.NDArray[np.float64]) -> None:
+        assert matrix.shape == (
+            3,
+            3,
+        ), "Transformation matrix must be a homogenous 3x3 matrix"
+        self._matrix = matrix
+
+    def translation(self) -> Vector2:
+        """Returns only the translation that this Transform applies
+
+        Returns:
+            Vector2: translation
+        """
+        m = self._matrix[:, 2]
+        m = m / m[2]
+        return Vector2(m)
+
+    def rotation(self) -> float:
+        """Returns only the rotation that this Transform applies
+
+        Returns:
+            float: rotation angle in radians
+
+            - angle > 0: CCW
+            - angle < 0: CW
+        """
+        # Not the most efficient solution
+        v = Vector2.new(1.0, 0.0)
+        v_rot: Vector2 = self * v
```

Comment:
it should be np.arctan2(self._matrix[1,0], self._matrix[0, 0]) 
Dont know if we really need this piece of performance